core/common/src/Instant.kt

```diff
@@ -123,12 +123,18 @@ public expect class Instant : Comparable<Instant> {
 
         /**
          * Parses a string that represents an instant in ISO-8601 format including date and time components and
-         * the mandatory `Z` designator of the UTC+0 time zone and returns the parsed [Instant] value.
+         * the mandatory time zone offset and returns the parsed [Instant] value.
          *
-         * Examples of instants in ISO-8601 format:
+         * In addition to allowing the `Z` designator of the UTC+0 time zone to be passed as the time zone offset, which
+         * is required by the ISO-8601, this parser also accepts the other possible offsets, but does not provide a way
+         * to query the result for which offset was specified in the string.
```

Comment:
I propose to rephrase this simpler, something like:
> Supports the following ways of specifying the time zone offset:
> - `Z` designator for the UTC+0 time zone,
> - a custom time zone offset specified with `+hh`, or `+hh:mm`, or `+hh:mm:ss`, where negative offsets are prefixed with `-` instead of `+`.

Reply:
Agreed, I was clever by half there. However, now that we are so explicit about the accepted offsets, we can't just ignore that parsing `+hh` fails on JS, so I added a workaround for that.

Reply:
Turns out that JDK8 also fails to parse `+hh`, as evidenced by the build failing. Added a workaround for that, too.